### ISSUE TITLE
Pagination should allow fetching pages in batch of N pages.

### ISSUE THREAD

**Issue by yantrikskaran** [googleapis/google-cloud-python#10430]
Currently I am facing an issue where I have to process big number of records coming as part of query result. The results are available in multiple pages.

When I am trying to process the row from result I notice that it takes around 10 seconds to load next page. I am able to process one page in a sec or 2 and then I have to wait for 10 seconds for next page. It is time consuming.

I cannot load entire result in memory as the response can be more than 10 GB and we start hitting various QUOTA limits.

It there a way i can load batch of 100 pages in memory at once and start my process and in the background next batch of 100 pages are loaded during processing on first batch of pages.

**Comment by plamut** [comment]
Just to clarify, which client library is this question about? BigQuery?

**Comment by yantrikskaran** [comment]
> Just to clarify, which client library is this question about? BigQuery?

yes it is regarding Big Query and I am using google.cloud.bigquery.client. 

Just ass more details - below is the snippet of my code.
    client = bigquery.Client()
    select_query = client.query(query)
    result = select_query.result()
    for row in result:
         print(row)     

# After printing around 1000 records I can see next record is has a gap of 10 sec. Looks like each page has around 1000 records and when it is fetching next page it is take time.

LOGS - Check below logs. Time before and after ==== line.

2020-05-18 03:33:36,380 MainThread (ProcessId : 7056) INFO create_tfRecords.py:(78) Completed writing to file
2020-05-18 03:33:36,380 MainThread (ProcessId : 7056) INFO util.py:(86) Memory Used By Process : 1000.03 MB
=======================================
2020-05-18 03:33:47,691 MainThread (ProcessId : 7056) INFO create_tfRecords.py:(150) Start : 157053
2020-05-18 03:33:47,691 MainThread (ProcessId : 7056) INFO create_tfRecords.py:(210) year_week : 201931
